aws-opentelemetry-distro/src/amazon/opentelemetry/distro/aws_opentelemetry_distro.py

```diff
@@ -36,6 +40,21 @@ def _configure(self, **kwargs):
             OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION environment variable. Need to work with upstream to
             make it to be configurable.
         """
+
+        # related issue: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2495
+        # mimicking what is done here: https://tinyurl.com/54mvzmte
+        # For handling applications like django running in containers, we are setting the current working directory
+        # to the sys.path for the django application to find its executables.
+        #
+        # Note that we are updating the sys.path and not the PYTHONPATH env var, because once sys.path is
+        # loaded upon process start, it doesn't refresh from the PYTHONPATH value.
```

Comment:
Not totally sure I understand this - why does the upstream [change](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/0a231e57f9722e6101194c6b38695addf23ab950/opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/__init__.py#L98-L102) not have to worry about this?

Reply:
Did some more digging to understand why the Otel code works by s[etting the CWD in the ](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/__init__.py#L98-L102)`PYTHONPATH` but in ADOT we have to set the CWD in the `sys.path`.
The full command to run auto-instrumentation for on-prem applications is `opentelemetry-instrument python app.py`. Here, the `opentelemetry-instrument` runs first, sets the CWD in `PYTHONPATH` and then the `python app.py` invokes a new python process. When the new process starts, the `PYTHONPATH` entries are populated in the `sys.path`.
When not using the `opentelemetry-instrument` command (such as in K8s) the auto-instrumentation starts with [sitecustomize.py](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/sitecustomize.py) as the entry-point, the `python app.py` process has already started and the `sys.path` has already read the `PYTHONPATH`. So we need to update the `sys.path` directly as `PYTHONPATH` mutation has no effect on the `sys.path` within the same process.